lib/LongVectorLoweringPass.cpp

```diff
@@ -1546,6 +1559,213 @@ LongVectorLoweringPass::convertBuiltinCall(CallInst &VectorCall,
                                 ScalarFactory);
 }
 
+Value *LongVectorLoweringPass::convertAllBuiltinCall(
+    CallInst &CI, Type *EquivalentReturnTy, ArrayRef<Value *> EquivalentArgs) {
+  Function *Builtin = CI.getCalledFunction();
+  assert(Builtin);
+  const auto &Info = clspv::Builtins::Lookup(Builtin);
+
+  switch (Info.getType()) {
+  default:
+    return convertBuiltinCall(CI, EquivalentReturnTy, EquivalentArgs);
+  case clspv::Builtins::kShuffle:
+    return convertBuiltinShuffle(CI, EquivalentReturnTy, EquivalentArgs);
+  case clspv::Builtins::kShuffle2:
+    return convertBuiltinShuffle2(CI, EquivalentReturnTy, EquivalentArgs);
+  }
+}
+
+Value *LongVectorLoweringPass::convertBuiltinShuffle(
```

Comment:
Because we cannot extractvalue from a variable index, I used a kind of dirty trick to do it which is to allocate on the stack and copy the vector, in order to have something where I can access with a variable index.
For shuffle, I have 1 alloc, while for shuffle2 I have to make 2 of them. Of course it would work to map shuffle on shuffle2 as you propose, but it would mean that shuffle would not perform as efficiently as it would with this implementation. Not sure if it is really a big deal.
I can do the modification if you want, but just wanted to make sure that you had all the information.

Reply:
I understand, but this could be handled by reusing `src` and in the shuffle2 alias the first alloca if `src1` and `src2` are the same (could even use that same check and avoid the comparison).

Reply:
good idea, I will do that